### PR TITLE
[moxie] Adding Strategy For Moxie Voting Power Calculation

### PR DESCRIPTION
**Pull request by Hrishikesh-Thakkar**
Changes proposed in this pull request:
- Adding a custom strategy for calculating moxie voting power which involves multiple subgraphs and RPC Calls for Moxie Token
- Adding Moxie as a strategy in the list of strategies

### CODE REVIEW

File: src/strategies/moxie/index.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+export const author = 'Hrishikesh-Thakkar';
+export const version = '0.0.1';
+
+const abi = [
+  'function balanceOf(address account) external view returns (uint256)'
+];
+
+const MOXIE_VESTING_SUBGRAPH_URL = "https://api.studio.thegraph.com/query/88457/moxie-vesting/version/latest";
+const MOXIE_PROTOCOL_SUBGRAPH_URL = "https://api.studio.thegraph.com/query/88457/moxie-protocol/version/latest";
+const MOXIE_LIQUIDITY_POOL_SUBGRAPH_URL = "https://api.studio.thegraph.com/query/88457/moxie-liquidity/version/latest";
```
Comment (ChaituVR) [inline src/strategies/moxie/index.ts:15]: Hi @Hrishikesh-Thakkar These URLs are not meant for production use, could you publish and share the URLs of the subgraphs? i can update the code with our proxy server

File: src/strategies/moxie/index.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      allAddressesScoreMap[getAddress(address)] += formattedBalance;
+    });
+  };
+
+  // Fetch data from both subgraphs in parallel
+  await Promise.all([
+    fetchSubgraphData(MOXIE_PROTOCOL_SUBGRAPH_URL, protocolSubgraphQuery, processProtocolData, "portfolios"),
+    fetchSubgraphData(MOXIE_LIQUIDITY_POOL_SUBGRAPH_URL, liquidityPoolSubgraphQuery, processLiquidityPoolData, "userPools"),
+    fetchBalances(),
+  ]);
```
Comment (ChaituVR) [inline src/strategies/moxie/index.ts:177]: It looks like the strategy is crossing the limit of 5 API calls/multicalls https://github.com/snapshot-labs/snapshot-strategies?tab=readme-ov-file#code You see some other way around?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/moxie/index.ts (hunk 2 above)
Comment (Hrishikesh-Thakkar) [reply src/strategies/moxie/index.ts:177]: Ah I thought the 5 requests included pagination. So would fetching all balances in a single multi-call count as 1 call?

Comment (ChaituVR) [reply src/strategies/moxie/index.ts:177]: Yep

File: src/strategies/index.ts [hunk 3]
```diff
@@ -923,8 +924,10 @@ const strategies = {
   pom,
   superboring,
   'erable-governance-v1': erableGovernanceV1,
+  'snx-multichain': snxMultichain,
```
Comment (ChaituVR) [inline src/strategies/index.ts:927]:
```suggestion
```

Comment (Hrishikesh-Thakkar) [reply src/strategies/index.ts]: Done 👍

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**